music21/test/test_expressions.py

```diff
@@ -161,11 +163,400 @@ def testTrillExtensionA(self):
+        self.assertEqual(turn.ornamentalPitches[1].accidental, pitch.Accidental('natural'))
+        self.assertTrue(turn.ornamentalPitches[1].accidental.displayStatus)
+        past = [pitch.Pitch('a#3'), pitch.Pitch('c#'), pitch.Pitch('c')]
+        turn.updateAccidentalDisplay(pitchPast=past, cautionaryPitchClass=False)
+        self.assertEqual(turn.ornamentalPitches[0], expectedOrnamentalPitches[0])
+        self.assertEqual(turn.ornamentalPitches[0].accidental, pitch.Accidental('natural'))
+        self.assertTrue(turn.ornamentalPitches[0].accidental.displayStatus)
+        self.assertEqual(turn.ornamentalPitches[1], expectedOrnamentalPitches[1])
+        self.assertEqual(turn.ornamentalPitches[1].accidental, pitch.Accidental('natural'))
+        self.assertTrue(turn.ornamentalPitches[1].accidental.displayStatus)
+
+    def testEdgeCases(self):
+        # Make sure you can call resolveOrnamentalPitches() on non-Trill/Mordent/Turn Ornaments
+        # without raising an exception (or actually doing anything interesting).
+        orn = expressions.Ornament()
+        orn.resolveOrnamentalPitches(note.Note('C4'))
+        self.assertEqual(orn.ornamentalPitches, tuple())
+
+        # Make sure you can call realize() on non-Trill/Mordent/Turn ornaments and
+        # just get an identical note back, with empty before and after lists.
+        n = note.Note('C4')
+        realized = orn.realize(n, inPlace=False)
+        self.assertEqual(realized, ([], n, []))
+        self.assertIsNot(realized[1], n)
+        realized = orn.realize(n, inPlace=True)
+        self.assertEqual(realized, ([], n, []))
+        self.assertIs(realized[1], n)
+
+        # Make sure that fillListOfRealizedNotes raises if srcObj is Unpitched, but
+        # transposeInterval is not unison.
+        filledList = []
+        with self.assertRaises(TypeError):
+            orn.fillListOfRealizedNotes(
+                note.Unpitched(),
+                fillObjects=filledList,
+                transposeInterval=interval.Interval('M2')
+            )
+        self.assertEqual(filledList, [])
+
+        # Make sure various Ornaments have correct direction
+        self.assertEqual(expressions.Trill().direction, 'up')
+        self.assertEqual(expressions.InvertedTrill().direction, 'down')
+        self.assertEqual(expressions.Mordent().direction, 'down')
+        self.assertEqual(expressions.InvertedMordent().direction, 'up')
+
+        # Make sure Turn.getSize raises on bad which value (must be 'upper' or 'lower')
+        with self.assertRaises(expressions.ExpressionException):
+            expressions.Turn().getSize(note.Note('C4'), which='bad: not upper or lower')
+
+    def testUnpitchedOrnaments(self):
         unp = note.Unpitched()
+
         mord = expressions.Mordent()
-        with self.assertRaises(TypeError):
-            mord.realize(unp)  # type: ignore
+        realized = mord.realize(unp)  # type: ignore
```

Comment:
Do you think we should loosen the typing of the `srcNote` arg to `realize` and `fillListOfRealizedNotes` to `Note | Unpitched`?

Reply:
Yes.  Done.

Reply:
Does this still need type: ignore now?

Reply:
No. Fixed that.